src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/UserDefinedExplicitConversions.cs

```diff
@@ -222,27 +222,27 @@ private void AddUserDefinedConversionsToExplicitCandidateSet(
                 isExplicit ? (isChecked ? WellKnownMemberNames.CheckedExplicitConversionName : WellKnownMemberNames.ExplicitConversionName) : WellKnownMemberNames.ImplicitConversionName,
                 operators);
 
-            var candidates = ArrayBuilder<MethodSymbol>.GetInstance(operators.Count);
-            candidates.AddRange(operators);
-
             if (isExplicit && isChecked)
             {
-                var operators2 = ArrayBuilder<MethodSymbol>.GetInstance();
-                declaringType.AddOperators(WellKnownMemberNames.ExplicitConversionName, operators2);
-
                 // Add regular operators as well.
                 if (operators.IsEmpty)
                 {
-                    candidates.AddRange(operators2);
+                    declaringType.AddOperators(WellKnownMemberNames.ExplicitConversionName, operators);
                 }
                 else
                 {
+                    var originalOperators = ArrayBuilder<MethodSymbol>.GetInstance(operators.Count);
```

Comment:
Consider, instead of allocating and copying here, just saving `operators.Count` here, and switching to `for` loop below.

Reply:
Nice idea!